File: fix_errors.mjs
import fs from 'fs';
import { globSync } from 'glob';

const files = globSync('src/**/*.{tsx,ts}');

for (const file of files) {
  let content = fs.readFileSync(file, 'utf8');
  let changed = false;

  // Replace {error} inside JSX text nodes
  const newContent = content.replace(/>\s*\{error\}\s*</g, ">{typeof error === 'string' ? error : JSON.stringify(error)}<");
  if (newContent !== content) {
    content = newContent;
    changed = true;
  }
  
  // Replace {data.error}
  const newContent2 = content.replace(/>\s*\{data\.error\}\s*</g, ">{typeof data.error === 'string' ? data.error : JSON.stringify(data.error)}<");
  if (newContent2 !== content) {
    content = newContent2;
    changed = true;
  }
  
  // Replace {applyError.error}
  const newContent3 = content.replace(/>\s*\{applyError\.error\}\s*</g, ">{typeof applyError.error === 'string' ? applyError.error : JSON.stringify(applyError.error)}<");
  if (newContent3 !== content) {
    content = newContent3;
    changed = true;
  }

  if (changed) {
    fs.writeFileSync(file, content);
    console.log(`Updated ${file}`);
  }
}

File: fix_errors.py
import os
import re

for root, _, files in os.walk("src"):
    for file in files:
        if file.endswith(".ts") or file.endswith(".tsx"):
            filepath = os.path.join(root, file)
            with open(filepath, "r") as f:
                content = f.read()
            
            # Replace console.error(error) and similar
            new_content = re.sub(r'console\.error\(\s*(e|error|err)\s*\)', r'console.error(\1?.message || \1)', content)
            
            # Replace console.error("...", error) 
            new_content = re.sub(r'console\.error\(\s*("[^"]*")\s*,\s*(e|error|err)\s*\)', r'console.error(\1, \2?.message || \2)', content)

            if new_content != content:
                with open(filepath, "w") as f:
                    f.write(new_content)
